tldw_Server_API/tests/Services/test_drain_gate_middleware.py

```
@@ -0,0 +1,147 @@
+
+    blocked = draining_client.post("/api/v1/chat/completions", json={"messages": []})
+    if blocked.status_code != 503:
+        raise AssertionError(f"expected drain gate to return 503, got {blocked.status_code}")
+    if blocked.json()["reason"] != "shutdown_in_progress":
+        raise AssertionError(f"unexpected drain reason: {blocked.json()['reason']!r}")
+    if budget_called["value"]:
+        raise AssertionError("LLMBudgetMiddleware was invoked for a drained request")
+
+
+@pytest.mark.parametrize(
+    "method,path",
```

Ruff (0.15.7)

[warning] Wrong type passed to first argument of `pytest.mark.parametrize`; expected `tuple`

Use a `tuple` for the first argument

(PT006)